compiler/src/typed/module_resolution.re

```diff
@@ -202,30 +202,65 @@ let locate_module = (~disable_relpath=false, base_dir, path, unit_name) => {
   ) {
   | Some(m) => m
   | None =>
-    let grain_src_exts = [".gr"];
     let (dir, m) =
-      switch (
-        find_in_path_uncap(~exts=[".gr.wasm"], base_dir, path, unit_name)
-      ) {
-      | (objpath, dir, basename, ext) =>
+      switch (find_in_path_uncap(~check_wasm=true, base_dir, path, unit_name)) {
+      | (objpath, dir, basename) =>
         ignore(log_resolution(unit_name, dir, basename));
-        switch (find_ext_in_dir(dir, basename, grain_src_exts)) {
-        | Some((srcpath, _, _, _)) => (
+        let file = find_ext_in_dir(dir, basename, [""]);
+        switch (file) {
+        | Some((srcpath, _, _)) => (
             dir,
             GrainModule(srcpath, Some(objpath)),
           )
         | None => (dir, WasmModule(objpath))
         };
       | exception Not_found =>
-        let (srcpath, dir, _, _) =
-          find_in_path_uncap(~exts=grain_src_exts, base_dir, path, unit_name);
+        let (srcpath, dir, _) =
+          find_in_path_uncap(~check_src=true, base_dir, path, unit_name);
         (dir, GrainModule(srcpath, None));
       };
     PathTbl.add(current_located_module_cache(), (dir, unit_name), m);
     m;
   };
 };
 
+let try_locate_module =
+    (~disable_relpath=false, base_dir, active_search_path, name, loc) => {
+  let locate = locate_module(~disable_relpath, base_dir, active_search_path);
+  Filepath.String.(
+    try(locate(name)) {
+    | Not_found =>
+      if (check_suffix(name, ".gr")) {
+        let no_extension = chop_suffix(name, ".gr");
+        switch (locate(no_extension)) {
+        | exception Not_found => error(No_module_file(loc, name, None))
+        | _ =>
+          let name = !is_relpath(name) ? no_extension : name;
+          error(
+            No_module_file(
+              loc,
+              name,
+              Some("did you forget to remove the .gr extension?"),
```

Comment:
I don't love this error message because it implies that there was something they were supposed to do but forgot, when really there wasn't a local file matching the name, but there is one on the module search path, and maybe that's what they wanted to reference.

Maybe something like `` "path/to/module.gr" was not found locally. Did you mean "path/to/module"? ``